src/ui/App.jsx

```diff
@@ -52,15 +56,57 @@ class App extends Component {
 
   componentWillMount() {
     const { cookies } = this.props;
+    this.getAuthCookiesAndSetAuthState();
```

Comment:
If I'm not mistaken `componentWillMount` is called before `render` method being called and setting state here won't cause a re-render. Calling `setState` anywhere inside the function body here is basically equivalent to changing the value of a variable, without causing any React side-effects.

That said I did change the code here a little bit to improve the quality, but I don't think one of these functions should be called as a call-back to other one because a) the order of execution here is not important and b) one function has an async call in it -- where the `setState` is called, and delaying the second function here for that call -- which may fail in fact, does not look very appealing to me.

Reply:
You're quite right. Looks nicer now in any case.